pkg/container/kubernetes/client.go

```diff
@@ -246,17 +247,30 @@ func (c *Client) DeployWorkload(ctx context.Context,
 	}
 
 	// Ensure the pod template has required configuration (labels, etc.)
-	podTemplateSpec = ensurePodTemplateConfig(podTemplateSpec, containerLabels)
+	isOpenShift := false
+	// Get a config to talk to the apiserver
+	cfg, err := clientconfig.GetConfig()
```

Comment:
I wonder if we should use ` rest.InClusterConfig` here?

Reply:
Agreed, we have some examples here with [mkp](https://github.com/StacklokLabs/mkp/blob/main/pkg/k8s/client.go#L106).